streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -476,7 +485,23 @@ public StreamsConfig(final Map<?, ?> props) {
 
         return consumerProps;
     }
-
+                 
+    private void checkIfUnexpectedUserSpecifiedConsumerConfig(Map clientProvidedProps) {
```

Comment:
Nit: add `final` -> `(final Map clientProviderProps)` (in Streams code base, we try to use final whenever possible)

Add generics to `Map`